Makefile

```diff
@@ -122,6 +122,14 @@ $(TEST_TARGETS): run-tests
 
 test-unit-cover: ARGS=-timeout=15m -coverprofile=coverage.txt -covermode=atomic
 test-unit-cover: TEST_PACKAGES=$(PACKAGES_UNIT)
+test-unit-cover:
+	@echo "Running unit tests with coverage..."
+	@go test -tags=test -mod=readonly $(ARGS) $(EXTRA_ARGS) $(TEST_PACKAGES)
+	@echo "Filtering ignored files from coverage.txt..."
+	@grep -v -E '/cmd/|/client/|/proto/|/testutil/|/mocks/|/test_.*\.go:|\.pb\.go:|\.pb\.gw\.go:|/x/[^/]+/module\.go:|/scripts/|/ibc/testing/|/version/|\.md:|\.pulsar\.go:' coverage.txt > filtered_coverage.txt
+	@echo "Function-level coverage summary:"
+	@go tool cover -func=filtered_coverage.txt
```

Comment:
You should probably directly overwrite `coverage.txt` here because a lot tools assume the coverage file is at the location. 

If you prefer to keep it here though you should add it to `.gitignore`